src/lib/openjp3d/tcd.c

```diff
@@ -226,10 +226,10 @@ void tcd_malloc_encode(opj_tcd_t *tcd, opj_volume_t * volume, opj_cp_t * cp,
         if (tcp->rates[j] <= 1) {
             tcp->rates[j] = 0;
         } else {
-            float num = (float)(tile->numcomps * (tile->x1 - tile->x0) *
-                                (tile->y1 - tile->y0) * (tile->z1 - tile->z0) * volume->comps[0].prec);
-            float den = (float)(8 * volume->comps[0].dx * volume->comps[0].dy *
-                                volume->comps[0].dz);
+            float num = (float)tile->numcomps * (tile->x1 - tile->x0) *
+                                (tile->y1 - tile->y0) * (tile->z1 - tile->z0) * volume->comps[0].prec;
+            float den = (float)8 * volume->comps[0].dx * volume->comps[0].dy *
```

Comment:
Same comment and question as above. In any case the type cast is not necessary – writing `8.0f` is the same as `(float)8`.

Reply:
```suggestion
            float den = 8.0f * volume->comps[0].dx * volume->comps[0].dy *
```
Looks probably better.

About the overflowing part, it depends on how big the slices are and how much of them we use. So it can occur basically at any of the multiplications.

Anyway casting the first operand to (long long) works as well but this expression is evaluated only once during compression and once during decompression so I don't think that the impact on the performance is that significant. When comparing the execution times the difference is not really visible.

Reply:
@ichlubna  OpenJPEG should only require ANSI C (C89) compilers. The _long long_ is specified since the C99 version of the standard. Casting to _float_ is fine.